client/src/pages/Campaign.js

```diff
@@ -84,27 +76,83 @@ const Campaign = (props) => {
     return {};
   }
 
+  const createStepObject = stepData => {
+    return {
+      id : stepData.id,
+      templateId : stepData.email_template.id,
+      templateName : stepData.email_template.name,
+      sent : 100,
+      replied : 25
+    }
+  }
+
+  const handleCampaign = data => {
+    const campaign = data.campaign;
+    const stepsData = campaign.steps;
+    const steps = [];
+    for(let stepData of stepsData) {
+      const step = createStepObject(stepData);
+      steps.push(step);
+    }
+
+    setCampaign(
```

Comment:
Not sure if this is throwing an error? 
Maybe assign a variable to this function? const setCampaign = () =>